### DISCUSSION TITLE
How to add a WPLANG ?

### DISCUSSION THREAD

**Discussion by antoinegreuzard** [wordplate/wordplate#342] [Q&A] [answered]
How to add a WPLANG like French?
I think there is no french language pack, but maybe it is possible to add it?

I tried with loco translate and with polylang, but not working.
I added define( 'WPLANG', env( 'WPLANG', 'fr_FR' ) );, not working too.

**Comment by vinkla** [comment, marked as answer]
I suggest looking at Polylang.

**Reply by antoinegreuzard** [reply]
Hello, 

Thanks for your reply but i tried and it's not working :/
Any other solution?

**Reply by vinkla** [reply]
If you want to translate the dashboard:

https://github.com/wp-languages/wp-languages.github.io

**Reply by antoinegreuzard** [reply]
It's work! Thanks!

**Reply by Kingo11** [reply]
@vinkla did u remove the installation language selection on your project ?
Because i cannot choose my language on installation lik u said in this topic: [https://github.com/vinkla/wordplate/issues/194#issuecomment-364738473](https://github.com/vinkla/wordplate/issues/194#issuecomment-364738473)
My only solution is to use your previous link.

**Reply by antoinegreuzard** [reply]
@Kingo11 Use the https://github.com/wp-languages/wp-languages.github.io package
Delete wordpress/ in public and vendor/ root, after run composer update
Here my composer.json :
```
{
    "name": "vinkla/wordplate",
    "description": "The WordPlate boilerplate",
    "license": "MIT",
    "type": "project",
    "keywords": [
        "wordplate",
        "wordpress",
        "composer",
        "bedrock",
        "wp",
        "languages"
    ],
    "require": {
        "php": "^8.1",
        "composer/installers": "^2.2",
        "koodimonni-language/fi": "*",
        "koodimonni-language/fr_fr": "*",
        "koodimonni/composer-dropin-installer": "dev-master",
        "laravel/valet": "^3.3",
        "roots/bedrock-autoloader": "^1.0",
        "roots/wordpress": "^6.1.1",
        "roots/wp-password-bcrypt": "^1.1",
        "symfony/http-foundation": "^6.0",
        "vinkla/headache": "^2.4",
        "vlucas/phpdotenv": "^5.5",
        "wpackagist-plugin/classic-editor": "^1.6",
        "wpackagist-plugin/clean-image-filenames": "^1.4"
    },
    "require-dev": {
        "roave/security-advisories": "dev-latest",
        "symfony/var-dumper": "^6.0"
    },
    "repositories": [
        {
            "type": "composer",
            "url": "https://wpackagist.org",
            "only": [
                "wpackagist-plugin/*",
                "wpackagist-theme/*"
            ]
        },
        {
            "type": "composer",
            "url": "https://wp-languages.github.io",
            "only": [
                "koodimonni-language/*",
                "koodimonni-plugin-language/*",
                "koodimonni-theme-language/*"
            ]
        }
    ],
    "autoload": {
        "files": [
            "src/helpers.php"
        ]
    },
    "config": {
        "allow-plugins": {
            "composer/installers": true,
            "roots/wordpress-core-installer": true,
            "koodimonni/composer-dropin-installer": true
        },
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "extra": {
        "installer-paths": {
            "public/mu-plugins/{$name}": [
                "type:wordpress-muplugin",
                "vinkla/headache",
                "wpackagist-plugin/clean-image-filenames"
            ],
            "public/plugins/{$name}": [
                "type:wordpress-plugin"
            ],
            "public/themes/{$name}": [
                "type:wordpress-theme"
            ]
        },
        "wordpress-install-dir": "public/wordpress",
        "dropin-paths": {
            "public/languages/": [
                "type:wordpress-language"
            ],
            "public/languages/plugins/": [
                "vendor:koodimonni-plugin-language"
            ],
            "public/languages/themes/": [
                "vendor:koodimonni-theme-language"
            ]
        }
    },
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ]
    }
}

```

**Reply by Kingo11** [reply]
@antoinegreuzard, @vinkla yes i know this solution, thanks :) but in my previous projects using Wordplate, i remember that i dont have to do that, just set the language at the installation like the old topic i mentioned, so i just want to know if something change on Wordplate or is it due to my environment. Thanks for the great work on Wordplate !

**Reply by vinkla** [reply]
We changed from using `johnpbloch/wordpress` to `roots/wordpress` without the content directory. Not sure if it could have an effect on the language picker.

**Reply by Kingo11** [reply]
@vinkla yep i just see your changes and i don't understand why i dont have the language picker on your project because i have it when i install bedrock..maybe my environment has a problem, i don't know. I tried on an other device and same.
Can you tell me if you ever had the language picker on your project on installation ?
Thanks

**Reply by vinkla** [reply]
You're right. Tried it locally and it seems `DISALLOW_FILE_MODS=true` disables the language picker. I haven't thought about it since I always build the dashboard in English. 

https://github.com/vinkla/wordplate/blob/3b2d76c13e8f4c14b6f1672507e742af3b9ef3aa/public/wp-config.php#L69-L70

**Reply by vinkla** [reply]
Added link to WP Languages in the documentation: https://github.com/vinkla/wordplate/commit/1d303685b17cf80f803e07a2edf26f0a1d89fb4c

**Reply by Kingo11** [reply]
> You're right. Tried it locally and it seems `DISALLOW_FILE_MODS=true` disables the language picker. I haven't thought about it since I always build the dashboard in English. 
> 
> 
> 
> https://github.com/vinkla/wordplate/blob/3b2d76c13e8f4c14b6f1672507e742af3b9ef3aa/public/wp-config.php#L69-L70

Oh nice, that's it. Thanks a lot !
